main.go

```diff
@@ -190,14 +197,48 @@ func NewServer() (*Server, error) {
 
 	bgpServer := bgpserver.NewBgpServer()
 
-	return &Server{
+	datastoreType := config.Spec.DatastoreType
+	server := Server{
 		bgpServer: bgpServer,
+		datastore: datastoreType,
 		client:    calicoCli,
-		etcd:      etcdCli,
 		ipv4:      ipv4,
 		ipv6:      ipv6,
 		reloadCh:  make(chan []*bgptable.Path),
-	}, nil
+	}
+
+	if datastoreType == calicoapi.EtcdV2 {
+		etcdConfig, err := getEtcdConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		cli, err := etcd.New(etcdConfig)
+		if err != nil {
+			return nil, err
+		}
+		server.etcd = etcd.NewKeysAPI(cli)
+	} else if datastoreType == calicoapi.Kubernetes {
+		k8s, err := NewK8sClient(&server)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Shouldn't this be "return nil, err"?

Reply:
Fatal level will call os.Exit(1).
https://godoc.org/github.com/sirupsen/logrus#FatalLevel

Reply:
So does "return nil, err".

Reply:
Ah, it was my misunderstanding. I will fix it.

Reply:
Done.